integrations/datadog/client.py

```diff
@@ -233,6 +296,373 @@ async def get_slo_history(
+            logger.error(f"Rate limit exceeded: {result.get('error')}")
+            return False
+        else:
+            logger.error(f"Error fetching data: {result.get('error')}")
+            return False
+
+    def create_query_with_values(
+        self, metric_query: str, variable_values: dict[str, str]
+    ) -> str:
+        """
+        Creates a Datadog query by substituting template variables with their values.
+
+        Args:
+            metric_query (str): The original query string containing template variables (e.g., "avg:container.cpu.usage{$service}").
+            variable_values (dict): A dictionary mapping variable names to their values (e.g., {"service": "svc_name", "env": "prod"}).
+
+        Returns:
+            str: The modified query with variables replaced by their values.
+        """
+        logger.debug(f"Creating query with values: {metric_query} | {variable_values}")
+        for variable_name, value in variable_values.items():
+            metric_query = metric_query.replace(
+                f"${variable_name}", f"{variable_name}:{value}"
+            )
+        return metric_query
+
+    def extract_queries_with_template_variable(
+        self, dashboard_json: dict[str, Any], variable_name: str
+    ) -> list[dict[str, Any]]:
+        """
+        Extracts metric queries from a Datadog dashboard JSON that use a specified template variable, grouping queries by widget.
+
+        Args:
+            dashboard_json (dict): The dashboard data as a Python dictionary.
+            variable_name (str): The name of the template variable to search for (e.g., "service").
+
+        Returns:
+            list: A list of dictionaries, each containing the widget ID, widget title, and a list of queries.
+        """
+        filter_name = f"${variable_name}"
+        matching_queries = {}
+
+        for widget in dashboard_json["widgets"]:
+            queries = [
+                query["query"]
+                for request in widget["definition"].get("requests", [])
+                for query in request.get("queries", [])
+                if filter_name in query["query"]
+            ]
+
+            if queries:
+                matching_queries[widget["id"]] = {
+                    "widget_id": widget["id"],
+                    "widget_title": widget["definition"].get("title", "Untitled"),
+                    "queries": queries,
+                }
+
+        return list(matching_queries.values())
+
+    async def get_single_dashboard(self, dashboard_id: str) -> dict[str, Any] | None:
+        if not dashboard_id:
+            return None
+        url = f"{self.api_url}/api/v1/dashboard/{dashboard_id}"
+        return await self._send_api_request(url)
+
+    async def enrich_dashboard_url(self, dashboard: dict[str, Any]) -> dict[str, Any]:
+        """
+        Enriches a dashboard item with the full Datadog web URL.
+        """
+        dashboard["url"] = f"{self.datadog_web_url}{dashboard['url']}"
```

Comment:
we can construct the url using the solution I shared in Slack: where we can use jq to access .env.INTEGRATION_VARIABLE